app/assets/javascripts/application.js

```diff
@@ -338,6 +338,26 @@ $(function() {
   });
 });
 
+function check_for_unavailable_puppetclasses() {
+  var unavailable_classes = $('#puppet_klasses #selected_classes .unavailable');
```

Comment:
Yes, I am also in favour of rewriting it under webpack, but wouldn't it be better to to have a component for the edit form in general first and then move over components like in this case for editing Puppet classes?

To have this tested now we can use an integration test, which can be removed (or not) once it is rewritten with tests under webpack.